Makefile

```diff
@@ -135,6 +143,30 @@ local-cross: bin/skopeo.darwin.amd64 bin/skopeo.linux.arm bin/skopeo.linux.arm64
 build-container:
 	${CONTAINER_RUNTIME} build ${BUILD_ARGS} -t "$(IMAGE)" .
 
+# Build container specific to arch
+build-arch-container:
+	${CONTAINER_RUNTIME} build -t "${ARCH_IMAGE}" contrib/skopeoimage/${SOURCE_TYPE}
+
+# Push container specific to arch
+push-arch-container:
+	echo "${QUAY_PASSWORD}" | ${CONTAINER_RUNTIME} login ${REGISTRY} -u "${QUAY_USERNAME}" --password-stdin
```

Comment:
I’m not too happy about this in the `Makefile`, especially when it fails in non-obvious ways if run without all the variables set.

If it makes sense to split code between Travis and `Makefile`, the `Makefile` targets should probably be useful for interactive use, in which case the user is presumably already logged in.

Would it be better placed in the Travis automation? I’m not sure at all.

---

`QUAY_{USERNAME,PASSWORD}` vs. `REGISTRY` is inconsistent.

To me, that’s another reason to move the login into Travis, and have Travis set full `…_IMAGE` values instead of individual components.

Reply:
> Would it be better placed in the Travis automation? I’m not sure at all.

Well, Travis has just 1 `.travis.yml` file to interact with project. Best practices here is to call make targets or just scripts from it, to avoid making it too complex. However I see your point here, we can have just separate directory with multi-arch build script there to call it from Travis. What do you think?

> To me, that’s another reason to move the login into Travis, and have Travis set full …_IMAGE values instead of individual components.

Still, I'd recommend to keep login in script, but yeah, probably env variables setup can be done in Travis

Reply:
I don’t have a strong opinion at all.

Would it make sense to split upstream build-automation scripts (compare `hack/travis_osx.sh`) from a `Makefile` targeted at all contributors running locally? I.e. move the `push-arch-container` task completely out of `Makefile` into `hack/travis_push_arch_container.sh` or something like that?

What do others prefer? @TomSweeneyRedHat ? @vrothberg ?

Reply:
actually I was thinking about having absolutely separate `release/Makefile` only for Travis images build. And to have in `release` directory also README.md with information about env variables, how build is done, etc..